cmake/GenerateConfigurationFile.cmake

```diff
@@ -67,18 +67,19 @@ check_struct_has_member("struct stat" st_mtim sys/stat.h
 check_struct_has_member("struct statfs" f_fstypename sys/mount.h
                         HAVE_STRUCT_STATFS_F_FSTYPENAME)
 
-include(CheckCXXCompilerFlag)
-
-# Old GCC versions don't have the required header support.
-# Old Apple Clang versions seem to support -mavx2 but not the target
-# attribute that's used to enable AVX2 for a certain function.
-if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
-   OR (CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang" AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 8.0))
-  message(STATUS "Detected unsupported compiler for HAVE_AVX2 - disabled")
-  set(HAVE_AVX2 FALSE)
-else()
-  check_cxx_compiler_flag(-mavx2 HAVE_AVX2)
-endif()
+include(CheckCXXSourceCompiles)
+check_cxx_source_compiles(
+  [=[
+    #include <immintrin.h>
+    void func() __attribute__((target("avx2")));
```

Comment:
Could you please explain why you need that attribute here? Theoretically a compiler could support AVX2 without also supporting the attribute.

Reply:
The reason I put the attribute test here is because the attribute is used in [src/hashutil.cpp](https://github.com/ccache/ccache/blob/master/src/hashutil.cpp#L114) and thus the compiler needs to support it to be able to build that file. In #689 there is a report of a compiler that supports -mavx2 but not the target attribute. As we are now checking for the actual support that is needed, I've removed the compiler specific tests that were here before.

The reason that the attribute is used instead of building the whole file with -mavx2 can be seen in [this commit](https://github.com/ccache/ccache/pull/476/commits/5043c1e5273a0e7e5f22b3ad940c16885ba7b2f8). I'm sure there are ways around that if someone really wants to.